google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineArtifactDeploymentSourceType.java

```diff
@@ -93,12 +85,16 @@ public void save(@NotNull ArtifactDeploymentSource deploymentSource,
       AppEngineDeployable deployable = (AppEngineDeployable) deploymentSource;
 
       if (deployable.getProjectName() != null) {
-        tag.setAttribute(PROJECT_ATTRIBUTE,
-            ((AppEngineDeployable) deploymentSource).getProjectName());
+        tag.setAttribute(PROJECT_ATTRIBUTE, deployable.getProjectName());
       }
 
       if (deployable.getVersion() != null) {
-        tag.setAttribute(VERSION_ATTRIBUTE, ((AppEngineDeployable) deploymentSource).getVersion());
+        tag.setAttribute(VERSION_ATTRIBUTE, deployable.getVersion());
+      }
+
+      if (deployable.getEnvironment() != null) {
```

Comment:
You can indeed remove this environment serialization as well. I've tested serializing / deserializing artifact deployment sources without this code and it still freshly gets the environment.